src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -195,63 +195,75 @@ bool mp::LXDVMImageVault::has_record_for(const std::string& name)
+    }
 }
 
 void mp::LXDVMImageVault::update_images(const FetchType& fetch_type, const PrepareAction& prepare,
                                         const ProgressMonitor& monitor)
 {
-    auto json_reply = lxd_request(manager, "GET", QUrl(QString("%1/images?recursion=1").arg(base_url.toString())));
-
-    auto images = json_reply["metadata"].toArray();
-
-    for (const auto image : images)
+    try
     {
-        auto image_info = image.toObject();
-        if (image_info.contains("update_source"))
+        auto json_reply = lxd_request(manager, "GET", QUrl(QString("%1/images?recursion=1").arg(base_url.toString())));
+        auto images = json_reply["metadata"].toArray();
+
+        for (const auto image : images)
         {
-            auto release = image_info["properties"].toObject()["release"].toString();
-            mpl::log(mpl::Level::debug, category, fmt::format("Checking if \'{}\' needs updating…", release));
+            auto image_info = image.toObject();
+            if (image_info.contains("update_source"))
+            {
+                auto release = image_info["properties"].toObject()["release"].toString();
+                mpl::log(mpl::Level::debug, category, fmt::format("Checking if \'{}\' needs updating…", release));
 
-            auto id = image_info["fingerprint"].toString();
+                auto id = image_info["fingerprint"].toString();
 
-            auto json_reply =
-                lxd_request(manager, "POST", QUrl(QString("%1/images/%2/refresh").arg(base_url.toString()).arg(id)));
+                auto json_reply = lxd_request(manager, "POST",
+                                              QUrl(QString("%1/images/%2/refresh").arg(base_url.toString()).arg(id)));
 
-            auto task_complete = [&release](auto metadata) {
-                if (metadata["metadata"].toObject()["refreshed"].toBool())
-                {
-                    mpl::log(mpl::Level::info, category, fmt::format("Image update for \'{}\' complete.", release));
-                }
-                else
-                {
-                    mpl::log(mpl::Level::debug, category, fmt::format("No image update for \'{}\'.", release));
-                }
-            };
+                auto task_complete = [&release](auto metadata) {
+                    if (metadata["metadata"].toObject()["refreshed"].toBool())
+                    {
+                        mpl::log(mpl::Level::info, category, fmt::format("Image update for \'{}\' complete.", release));
+                    }
+                    else
+                    {
+                        mpl::log(mpl::Level::debug, category, fmt::format("No image update for \'{}\'.", release));
+                    }
+                };
 
-            poll_download_operation(json_reply, monitor, task_complete);
+                poll_download_operation(json_reply, monitor, task_complete);
+            }
         }
     }
+    catch (const LXDNotFoundException&)
+    {
+        return;
+    }
```

Comment:
Ditto? There's a lot more requests than just the update one that you're catching here, and not all of them should be ignored, I'd say?

Reply:
Should be better now.